### PR TITLE
Fix can't archive a question from the Question page

### LINKED ISSUE #16512: Can't archive a question from the Question page
**Describe the bug**
When archiving a question from the question page, it navigates you to the collection as expected, but the item isn't achieved and there is a message saying "Unarchived the item"

**To Reproduce**
1. Open any question
2. Click the pencil icon near the question's name
3. Click archive
4. Confirm the archive
5. You're navigated to the collection page, but the question isn't achieved and there is a message saying "Unarchived the item"

**Expected behavior**
The item should disappear from the list and the toast message should say "Archived the item"

**Demo**

https://user-images.githubusercontent.com/17258145/121331325-717b3080-c91f-11eb-9e84-cecdb7c0dc7d.mov

**Information about your Metabase Installation:**

Metabase version: latest master ([commit](https://github.com/metabase/metabase/commit/130f73f292ad6f0cb81e1c6628285b36ec450921))
Metabase hosting environment: local, master branch
Metabase internal database: H2 (default), sample dataset

### PR DESCRIPTION
**Pull request by kulyk**
When archiving a question from the question page, it navigates you to the collection as expected, but the item isn't achieved and there is a message saying "Unarchived the item". Fixes #16512, reproduces #16512

There is [a test](https://github.com/metabase/metabase/blob/207cd9bce9e2399ffc6d5cfdc3d1db1f72b84d37/frontend/test/metabase/scenarios/collections/permissions.cy.spec.js#L304) that had to catch that. Looks like it was asserting the archived item is gone while the collection content was not loaded. There is a spinner in place of item lists, so the "not.exist" assertion was successful. Could make it work by explicitly waiting until collection items GET requests are done (there are two of them: one for pinned items and one for unpinned). Not sure if it's the best possible fix here

### To Verify

1. Open any question
2. Click the pencil icon near the question's name
3. Click archive and confirm the action
6. Ensure the question disappeared from the list and the toast message says "Archived the item", instead of "Unarchived"

### Demo

**Before**

https://user-images.githubusercontent.com/17258145/121331325-717b3080-c91f-11eb-9e84-cecdb7c0dc7d.mov

**After**

https://user-images.githubusercontent.com/17258145/121341746-a3919000-c929-11eb-8ac2-cc99eaad9ac6.mov

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
> There is a test that had to catch that. Looks like it was asserting the archived item is gone while the collection content was not loaded. There is a spinner in place of item lists, so the "not.exist" assertion was successful. Could make it work by explicitly waiting until collection items GET requests are done (there are two of them: one for pinned items and one for unpinned). Not sure if it's the best possible fix here.

Great catch, @kulyk and yes - perfectly good solution. Thanks for fixing this and sorry I missed it in the first place.

**Comment by kulyk** [comment]
> > There is a test that had to catch that. Looks like it was asserting the archived item is gone while the collection content was not loaded. There is a spinner in place of item lists, so the "not.exist" assertion was successful. Could make it work by explicitly waiting until collection items GET requests are done (there are two of them: one for pinned items and one for unpinned). Not sure if it's the best possible fix here.
> 
> Great catch, @kulyk and yes - perfectly good solution. Thanks for fixing this and sorry I missed it in the first place.

No worries, especially because I was the one who introduced the bug 😄

File: frontend/test/metabase/scenarios/collections/permissions.cy.spec.js [hunk 1]
```diff
@@ -302,9 +302,14 @@ describe("collection permissions", () => {
                 });
 
                 it("should be able to archive the question (metabase#11719-3)", () => {
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/collections/permissions.cy.spec.js:304]: Feel free to add the reference to #16512 as well.

Comment (kulyk) [reply frontend/test/metabase/scenarios/collections/permissions.cy.spec.js:304]: Does `(metabase#11719-3, metabase#16512)` look good?

**Review by nemanjaglumac — APPROVED** [review]
Excellent catch.
Works like a charm.

**Pull request merged**